### DISCUSSION TITLE
Attempted to load @next/swc-win32-ia32-msvc, but an error occurred

### DISCUSSION THREAD

**Discussion by warrengoldman** [vercel/next.js#61407] [Show and tell] [answered]
### Summary

v21.6.1 ia32 win32
Windows 10 Home Intel(R) Core(TM) i7-4510U CPU @ 2.00GHz   2.00 GHz
next: 10.4.0
I have node_modules\@next\swc-win32-ia32-msvc\next-swc.win32-ia32-msvc.node, whether it was downloaded correctly I am not sure how to tell that.

I followed all the steps noted on: https://nextjs.org/docs/messages/failed-loading-swc 

### Additional information

```js
>npm run dev

> dev
> next dev

   ▲ Next.js 14.1.0
   - Local:        http://localhost:3000

 ⚠ Attempted to load @next/swc-win32-ia32-msvc, but an error occurred: A dynamic link library (DLL) initialization routine failed.
\\?\C:\Users\TEST\eclipse-workspace\REACT\node_modules\@next\swc-win32-ia32-msvc\next-swc.win32-ia32-msvc.node
 ⨯ Failed to load SWC binary for win32/ia32, see more info here: https://nextjs.org/docs/messages/failed-loading-swc
```

**Comment by warrengoldman** [comment]
also attempted to disable swcMinify AND  ran this: npm i @next/swc-win32-ia32-msvc

still get same error (but other info in there). I did run npm install and npm run dev again but got same error

dev
> next dev

   ▲ Next.js 14.1.0
   - Local:        http://localhost:3000

 ⚠ Disabling SWC Minifer will not be an option in the next major version. Please report any issues you may be experiencing to https://github.com/vercel/next.js/issues
 ⚠ Attempted to load @next/swc-win32-ia32-msvc, but an error occurred: A dynamic link library (DLL) initialization routine failed.
\\?\C:\Users\TEST\eclipse-workspace\REACT\node_modules\@next\swc-win32-ia32-msvc\next-swc.win32-ia32-msvc.node
 ⚠ Found lockfile missing swc dependencies, patching...
 ⚠ Lockfile was successfully patched, please run "npm install" to ensure @next/swc dependencies are downloaded
 ⨯ Failed to load SWC binary for win32/ia32, see more info here: https://nextjs.org/docs/messages/failed-loading-swc

**Comment by warrengoldman** [comment, marked as answer]
The issue was I had node 32 and needed node 64.
Installing nvm really helped in making it easier to figure that out.

**Reply by MuhammadShoaib495** [reply]
Right!

**Reply by Rezwanul-Haque** [reply]
In my case, I installed the node using NVM on windows 11. 

`node -p process.arch`
it shows
`x64`

still getting the same error

**Comment by jaber1988** [comment]
Hey there!

I understand you're encountering problems with Node.js on your 64-bit system. It seems you might have a 32-bit version of Node.js installed, which can cause compatibility issues.

Here's how to resolve this:

1. Uninstall the 32-bit Node.js:

Open the Control Panel.
Go to Programs and Features (or Add or Remove Programs, depending on your Windows version).
Locate Node.js in the program list.
Right-click on it and select Uninstall.
Follow the on-screen prompts to complete the uninstallation.
2. Download and Install the 64-bit Node.js:

Head over to the official Node.js download page: https://nodejs.org/en/download
Under the LTS (Long-Term Support) section, download the Windows Installer (.msi) corresponding to the x64 architecture.
Double-click the downloaded installer and follow the on-screen instructions to complete the installation.
3. Verify the Installation:

Open a command prompt window (Start menu, search for cmd, press Enter).
Type node -v and press Enter. This should display the newly installed 64-bit Node.js version.
4. Optional: Restart Your Computer (Recommended):

While not strictly necessary, restarting your computer ensures any system-wide environment variables are updated to reflect the new Node.js installation.
Additional Tips:

Package Manager Compatibility: If you're using a package manager like npm or yarn, you may need to re-install your global packages after switching to the 64-bit version.
By following these steps, you should have Node.js 64-bit successfully installed and running on your system. If you encounter any further issues, feel free to provide more details, and the community can assist you further.

**Reply by Muz4mmil** [reply]
> Hey there!
> 
> I understand you're encountering problems with Node.js on your 64-bit system. It seems you might have a 32-bit version of Node.js installed, which can cause compatibility issues.
> 
> Here's how to resolve this:
> 
> 1. Uninstall the 32-bit Node.js:
> 
> Open the Control Panel. Go to Programs and Features (or Add or Remove Programs, depending on your Windows version). Locate Node.js in the program list. Right-click on it and select Uninstall. Follow the on-screen prompts to complete the uninstallation. 2. Download and Install the 64-bit Node.js:
> 
> Head over to the official Node.js download page: https://nodejs.org/en/download Under the LTS (Long-Term Support) section, download the Windows Installer (.msi) corresponding to the x64 architecture. Double-click the downloaded installer and follow the on-screen instructions to complete the installation. 3. Verify the Installation:
> 
> Open a command prompt window (Start menu, search for cmd, press Enter). Type node -v and press Enter. This should display the newly installed 64-bit Node.js version. 4. Optional: Restart Your Computer (Recommended):
> 
> While not strictly necessary, restarting your computer ensures any system-wide environment variables are updated to reflect the new Node.js installation. Additional Tips:
> 
> Package Manager Compatibility: If you're using a package manager like npm or yarn, you may need to re-install your global packages after switching to the 64-bit version. By following these steps, you should have Node.js 64-bit successfully installed and running on your system. If you encounter any further issues, feel free to provide more details, and the community can assist you further.

I have the same problem and i have a 32 bit OS with 32bit nodejs, what should i do?

**Comment by WendaoLee** [comment]
An alternative solution is trying to uninstall your `node_modules`,then run `npm install` to re-install all modules.This works for my situation:
1. I've developed Next.js application in my Windows10 before.
2. I need using proxy to fetch npm packages due to the network restriction in my regeion,and that's the causes of this error for me——As I suspected.

**Comment by Ryaaan4321** [comment]
I had a similar issue and in my case deleting the node_modules folder and the package-lock.json file then running npm install worked for me.
one more thing to address is that when I faced the issue, I was working with monorepos. I deleted the base files (node_modules and package-lock.json), and it worked
